Topic by info is either 'nick' or 'account.users.quakenet.org'

See info in whoban.patch

diff -r 63c8cdb5dc47 include/channel.h
--- a/include/channel.h	Tue Mar 23 13:48:36 2010 +0100
+++ b/include/channel.h	Tue Mar 23 13:56:29 2010 +0100
@@ -285,9 +285,10 @@
   struct Mode        mode;	   /**< This channels mode */
   unsigned int       marker;       /**< Channel marker */
   char               topic[TOPICLEN + 1]; /**< Channels topic */
-  char               topic_nick[NICKLEN + 1]; /**< Nick of the person who set
+  char               topic_who[ACCOUNTLEN > NICKLEN ? ACCOUNTLEN+1 : NICKLEN+1]; /**< Nick or account of the person who set
 						*  The topic
 						*/
+  int                topic_who_is_account; /**< 0 when topic_who is nick, 1 when an account */
   char               chname[1];	   /**< Dynamically allocated string of the 
 				     * channel name
 				     */
diff -r 63c8cdb5dc47 ircd/m_burst.c
--- a/ircd/m_burst.c	Tue Mar 23 13:48:36 2010 +0100
+++ b/ircd/m_burst.c	Tue Mar 23 13:56:29 2010 +0100
@@ -361,7 +361,8 @@
     /* clear topic set by netrider (if set) */
     if (*chptr->topic) {
       *chptr->topic = '\0';
-      *chptr->topic_nick = '\0';
+      *chptr->topic_who = '\0';
+      chptr->topic_who_is_account = 0;
       chptr->topic_time = 0;
       sendcmdto_channel_butserv_butone(&his, CMD_TOPIC, chptr, NULL, 0,
                                        "%H :%s", chptr, chptr->topic);
diff -r 63c8cdb5dc47 ircd/m_check.c
--- a/ircd/m_check.c	Tue Mar 23 13:48:36 2010 +0100
+++ b/ircd/m_check.c	Tue Mar 23 13:56:29 2010 +0100
@@ -378,7 +378,8 @@
     send_reply(sptr, SND_EXPLICIT | RPL_DATASTR, ":          Topic:: %s", chptr->topic);
 
     /* ..set by */
-    send_reply(sptr, SND_EXPLICIT | RPL_DATASTR, ":         Set by:: %s", chptr->topic_nick);
+    send_reply(sptr, SND_EXPLICIT | RPL_DATASTR, ":         Set by:: %s%s%s",
+      chptr->topic_who, chptr->topic_who_is_account ? "." : "", chptr->topic_who_is_account ? feature_str(FEAT_HIDDEN_HOST) : "");
 
     send_reply(sptr, SND_EXPLICIT | RPL_DATASTR, ":         Set at:: %s (%Tu)", myctime(chptr->topic_time), chptr->topic_time); 
   }
diff -r 63c8cdb5dc47 ircd/m_join.c
--- a/ircd/m_join.c	Tue Mar 23 13:48:36 2010 +0100
+++ b/ircd/m_join.c	Tue Mar 23 13:56:29 2010 +0100
@@ -283,7 +283,10 @@
 
     if (chptr->topic[0]) {
       send_reply(sptr, RPL_TOPIC, chptr->chname, chptr->topic);
-      send_reply(sptr, RPL_TOPICWHOTIME, chptr->chname, chptr->topic_nick,
+      send_reply(sptr, RPL_TOPICWHOTIME, chptr->chname,
+                 chptr->topic_who,
+                 chptr->topic_who_is_account ? "." : "",
+                 chptr->topic_who_is_account ? feature_str(FEAT_HIDDEN_HOST) : "",
 		 chptr->topic_time);
     }
 
diff -r 63c8cdb5dc47 ircd/m_topic.c
--- a/ircd/m_topic.c	Tue Mar 23 13:48:36 2010 +0100
+++ b/ircd/m_topic.c	Tue Mar 23 13:56:29 2010 +0100
@@ -65,7 +65,10 @@
    newtopic=ircd_strncmp(chptr->topic,topic,TOPICLEN)!=0;
    /* setting a topic */
    ircd_strncpy(chptr->topic, topic, TOPICLEN);
-   ircd_strncpy(chptr->topic_nick, cli_name(from), NICKLEN);
+   ircd_strncpy(chptr->topic_who,
+     IsAccount(from) ? cli_user(from)->account : cli_name(from),
+     IsAccount(from) ? ACCOUNTLEN : NICKLEN);
+   chptr->topic_who_is_account = IsAccount(from) ? 1 : 0;
    chptr->topic_time = ts ? ts : TStime();
    /* Fixed in 2.10.11: Don't propagate local topics */
    if (!IsLocalChannel(chptr->chname))
@@ -138,7 +141,10 @@
       else
       {
 	send_reply(sptr, RPL_TOPIC, chptr->chname, chptr->topic);
-	send_reply(sptr, RPL_TOPICWHOTIME, chptr->chname, chptr->topic_nick,
+	send_reply(sptr, RPL_TOPICWHOTIME, chptr->chname,
+                   chptr->topic_who,
+                   chptr->topic_who_is_account ? "." : "",
+                   chptr->topic_who_is_account ? feature_str(FEAT_HIDDEN_HOST) : "",
 		   chptr->topic_time);
       }
     }
diff -r 63c8cdb5dc47 ircd/s_err.c
--- a/ircd/s_err.c	Tue Mar 23 13:48:36 2010 +0100
+++ b/ircd/s_err.c	Tue Mar 23 13:56:29 2010 +0100
@@ -698,7 +698,7 @@
 /* 332 */
   { RPL_TOPIC, "%s :%s", "332" },
 /* 333 */
-  { RPL_TOPICWHOTIME, "%s %s %Tu", "333" },
+  { RPL_TOPICWHOTIME, "%s %s%s%s %Tu", "333" },
 /* 334 */
   { RPL_LISTUSAGE, ":%s", "334" },
 /* 335 */
